utils/utils.go

```diff
@@ -1215,6 +1232,25 @@ func GetWatchNamespaces() ([]string, error) {
 	return watchNamespaces, nil
 }
 
+// MergeTopologySpreadConstraints returns the union of all TopologySpreadConstraint lists. The order of lists passed into the
+// func, defines the importance.
+func MergeTopologySpreadConstraints(constraints ...[]corev1.TopologySpreadConstraint) []corev1.TopologySpreadConstraint {
```

Comment:
Customers could set [default constraints for the entire cluster](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/#cluster-level-default-constraints). But those defaults would be overridden by our default constraints. It'll be good to consider providing an option/way, so they can disable our defaults without having to override each constraint (using the relevant topology key)

Reply:
Added flag `.spec.topologySpreadConstraints.disableOperatorDefaults`. It defaults to false, so operator default TopologySpreadConstraints will be applied if the flag isn't specified. However, this also means upgrading the operator will cause brief downtime for the pod to apply our default constraints.